packages/fcl/src/wallet-provider-spec/user-signature.md

```diff
@@ -0,0 +1,266 @@
+# User Signature
+
+## Status
+
+- **Last Updated:** June 1st 2021
+- **Stable:** Yes
+- **Risk of Breaking Change:** Medium
+- **Compatibility:** `>= @onflow/fcl@0.0.70`
+
+# Overview and Introduction
+
+**Personally sign data via FCL Compatible Wallets**
+
+**FCL** now incldues **`signUserMessage()`** which allows for the sending of unencrypted message data to a connected wallet provider or service to be signed with a user's private key. 
+
+An application or service can verify a signature against a user's public key on the **Flow Blockchain** providing proof a user controls the account's private key.   
+
+**Use Cases**
+
+- **Authentication**: Cryptographically verify the ownership of a **Flow** account by signing a piece of data using a private key
+- **Improved Application Login**
+  - **Increased security**: Arguably more secure than proof of ownership by email/password
+  - **Simplified UX**: No application password required
+  - **Increased privacy**: No email or third party authentication service needed
+- **Message Validation**: Assuring that a message sent or received has not been tampered with
+- **Multisig contracts**
+- **Decentralised exchanges**
+- **Meta transactions**
+
+
+# Config and Authentication
+
+As a prerequisite, **FCL** is configured to point to the Wallet Provider's Authentication Endpoint.
+
+> During development (and on mainnet) FCL can be configured to use the wallet directly by
+> setting the **Challenge Handshake Url** to the wallet providers **Authentication Endpoint**
+> by configuring fcl like this `config().put("challenge.handshake", "https://my-awesome-wallet-provider.com/fcl/authenticate")`.
+
+1. A user initiates authentication with the wallet provider via application UI
+2. The wallet confirms a user's identity and sends back information used to configure **FCL** for future user actions in the application
+3. Included in the authentication response should be the provider's [Key Services](#) including a **`user-signature`** service for use with **`signUserMessage()`**
+
+
+# User Signature Service
+
+A User Signing Service's job is to receive a signable message from **FCL**, and send back a **`Composite Signature`**.
+
+A **`user-signature`** service should conform to the following:
+
+```javascript
+{
+  f_type: "Service",
+  f_vsn: "1.0.0",
+  type: "user-signature",
+  uid: "uniqueDedupeKey",
+  endpoint: "http://rawr.io",
+  method: "IFRAME/RPC",
+  uid: "____",
+  endpoint: "https://____",
+  method: "IFRAME/RPC", 
+  "id": "____", // wallets internal id for the user
+  data: {},   // included in body of user-signature request
+  params: {}, // included as query params on endpoint url
```

Comment:
Doubling up on some values here.
Would probably just say its a standard service, with methods for `IFRAME/RPC` and `HTTP/POST`
Maybe link to the [normalizer](https://github.com/onflow/flow-js-sdk/blob/master/packages/fcl/src/current-user/normalize/user-signature.js#L4-L14) which is the source of truth for this shape anyways.

Reply:
👍🏼